### PR TITLE
add config for vsphere csi image promotion

### PR DESCRIPTION
**Pull request by divyenpatel**
Add config for vsphere CSI driver image promotion

cc: @XudongLiuHarold @xing-yang

### CODE REVIEW

**Comment by divyenpatel** [comment]
/hold

wait for https://github.com/kubernetes/k8s.io/pull/7210 merge

File: registry.k8s.io/images/k8s-staging-csi-vsphere/generate.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    driver-windows-20h2-amd64
+    driver-windows-ltsc2022-amd64
+    syncer
+)
+
+for image in "${images[@]}"; do
+    echo "- name: ${image}"
+    echo "  dmap:"
+    gcloud container images list-tags \
+        "${repo}/$image" \
```
Comment (upodroid) [inline registry.k8s.io/images/k8s-staging-csi-vsphere/generate.sh:40]: Can we start publishing unified multi-arch/multi-os images with correct tags to `us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere`? Thanks

Comment (divyenpatel) [reply registry.k8s.io/images/k8s-staging-csi-vsphere/generate.sh:40]: here repo points to "gcr.io/cloud-provider-vsphere/csi/release" from where we need to migrate older images.
we do not have any image here - us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere

Comment (divyenpatel) [reply registry.k8s.io/images/k8s-staging-csi-vsphere/generate.sh:40]: @upodroid added this file on the PR registry.k8s.io/manifests/k8s-staging-csi-vsphere/promoter-manifest.yaml

**Comment by divyenpatel** [comment]
@upodroid 

We have started pushing images here 

us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere/ci

what should be the final image we need to put in our manifest files? `k8s-staging-images` in the manifest for released tag does not sound correct.


```
divyenp@C02Z5410LVDQ ~ % docker manifest inspect us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere/ci/driver:latest
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1367,
         "digest": "sha256:801f284d243cb9836cc9327ccbc11634cddd752026a4d499a406215e00fc2efd",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:682567cac60eac9986b61d8fed15e572e1f8fcfd05bb3e02500d769ab81ff199",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.6189"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:5814b73f8e8cab0ee074dcc3a89818fcd35ef0ecb67082a466d8be1f210b0c8f",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19042.1889"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:c20117cd98590b957e3ff9f30f4e5c91d67ecf8a4665a6ad174de0d7143a3927",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.20348.2655"
         }
      }
   ]
}
divyenp@C02Z5410LVDQ ~ % docker manifest inspect us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere/ci/syncer:latest
{
	"schemaVersion": 2,
	"mediaType": "application/vnd.docker.distribution.manifest.v2+json",
	"config": {
		"mediaType": "application/vnd.docker.container.image.v1+json",
		"size": 2691,
		"digest": "sha256:83f4297151b1e3f9a7885cd46002d0fe1eabde936a438762b67c236179178ca9"
	},
	"layers": [
		{
			"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
			"size": 17140337,
			"digest": "sha256:32818a0e2f0073d327491bc78606ce63875a6e8bc57f7ab55355756d7027f535"
		},
		{
			"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
			"size": 520,
			"digest": "sha256:aa834454e75303ffeb6b47d660b90a6e26cc852abe7e55f49ed2648f4790046a"
		},
		{
			"mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
			"size": 17989360,
			"digest": "sha256:f4ce9c4165b29ba94b0156f45daa9df1ef0d300bb5707bfbd1a57fad38e5b704"
		}
	]
}

```

**Comment by divyenpatel** [comment]
@upodroid 

Project cloud-provider-vsphere has been deleted. 

https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/3053

I think we can not migrate older release to new location. How do fix this?

cc: @xing-yang

**Comment by xing-yang** [comment]
Discussed with @upodroid.  He said we should update the pipeline to not insert a ci prefix. After that we can then promote the image and serve it [registry.k8s.io/csi-vsphere/driver](http://registry.k8s.io/csi-vsphere/driver) instead of [registry.k8s.io/csi-vsphere/ci/driver](http://registry.k8s.io/csi-vsphere/ci/driver).

**Comment by divyenpatel** [comment]
we are prefixing CI for images for main

```
# CI images
readonly CSI_IMAGE_CI=${BASE_IMAGE_REPO}/ci/driver
readonly SYNCER_IMAGE_CI=${BASE_IMAGE_REPO}/ci/syncer
```

for release tag we are prefixing image with 

```
# Release images
readonly CSI_IMAGE_RELEASE=${BASE_IMAGE_REPO}/release/driver
readonly SYNCER_IMAGE_RELEASE=${BASE_IMAGE_REPO}/release/syncer
```

https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/hack/release.sh#L31-L33

So are you saying we don't want to do this and keep all images under `us-central1-docker.pkg.dev/k8s-staging-images/csi-vsphere`

**Comment by upodroid** [comment]
That's correct, there should be no differences between the images. You should be able to promote any staging image to production right away.

**Comment by divyenpatel** [comment]
@upodroid I am not sure if this PR is required as source repository `gcr.io/cloud-provider-vsphere/csi/release` is no longer present and looks deleted.

cc: @xing-yang @XudongLiuHarold @sbueringer

**Comment by sbueringer** [comment]
We should push to the community owned staging registry and then promote from there to the community owned production registry.

If the plan was to promote from VMware owned gcr.io/cloud-provider-vsphere/csi/release to the community owned production registry. This was never a valid option

**Pull request closed without merging**